### DISCUSSION TITLE
About blockchain ctf, need some advices

### DISCUSSION THREAD

**Discussion by 0xDennnnny** [Cyfrin/security-and-auditing-full-course-s23#259] [General]
Hello, to whom concern

Recently I'm working on onlypwner ctf challenge. 
I completed the half of them (level 3 difficulty ones) 

but when it comes to level4, I stuck ... 
I can not figure out the answer as fast as before. 

I know that I definitely need  some knowledge of solidity, evm, related fields.
However I dont know where to start, a little confused and frustrated right now... 

If anyone could give me some advices that will be wonderful. 
I want to know what should I do to get some knowledge about them. 
and what could I do now to sharp my ctf skill ?

If anyone want to, like discuss about ctf or solve some challenge together, discussion together 
are also welcome. 
need some company :) 

Thank you. 🙏

**Comment by EngrPips** [comment]
Hello @0xDennnnny, I think you should first of all figure out why you couldn't solve the ctf challenge and then do some study to bridge your knowledge gap, then attempt the challenge again.

**Reply by 0xDennnnny** [reply]
Thanks for reply. 
Right now I actually don't really know why I couldn't solve the challenge ? 
I said I don't know because I tried all the methods I knew to solve it but failed.
So ... I must lack some knowledge! Like you said, I should so some study to bridge my knowledge gap. 
and I know that.

I think ... the thing bother me right now is I don't know where should I start to learn that knowledge.
For me, it's like I can't tell where is the key point to solve the challenge 

What I'm trying to do right now, is to find some audit example in solodit, try to see is there any similar exploit in the past. 
But... I don't know, it's seems not really useful right now...  I'm kind of lost direction.

Thanks anyway, really appreciate your reply.

**Comment by EngrPips** [comment]
What is the `ctf` about?

**Reply by 0xDennnnny** [reply]
oh, It's [Multisig](https://onlypwner.xyz/challenges/14) in onlypwner. 

( my current probably direction is : ecdsa signature, or msg.data manipulate ... or somthing that I don't understand yet 🤔

**Reply by EngrPips** [reply]
I advise that you read some articles or posts about `Multisig` and how they work, read up on `ECDSA` signatures, and read up on `msg.sender` in an Ethereum transaction. After you finish all of those studying, then attempt the `ctf`

**Reply by 0xDennnnny** [reply]
Thanks for the advices!! I'll try on that way. thank you.

**Reply by EngrPips** [reply]
You are welcome, Friend. Looking forward to hearing back from you on how things play out.
